Alias elimination internal error

This model is two resistors in series.

```julia
@variables t
@variables v1(t) v2(t) i1(t) i2(t)
eq = [
    v1 ~ sin(2pi*t)
    v1 - v2 ~ i1
    v2 ~ i2
    i1 ~ i2
]
@named sys = ODESystem(eq)
sys = structural_simplify(sys)
```
The error is:
```
julia> sys = structural_simplify(sys)
ERROR: AssertionError: r == 0
Stacktrace:
 [1] exactdiv(a::Int64, b::Int64)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\alias_elimination.jl:303
 [2] locally_structure_simplify!(adj_row::ModelingToolkit.CLILVector{Int64, Int64}, pivot_col::Int64, ag::ModelingToolkit.AliasGraph, may_eliminate::Bool)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\alias_elimination.jl:347
 [3] (::ModelingToolkit.var"#lss!#576"{ModelingToolkit.AliasGraph, ModelingToolkit.SparseMatrixCLIL{Int64, Int64}, Vector{Int64}})(ei::Int64)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\alias_elimination.jl:261
 [4] foreach(f::ModelingToolkit.var"#lss!#576"{ModelingToolkit.AliasGraph, ModelingToolkit.SparseMatrixCLIL{Int64, Int64}, Vector{Int64}}, itr::StepRange{Int64, Int64})
   @ Base .\abstractarray.jl:2141
 [5] alias_eliminate_graph!(graph::BipartiteGraph{Int64, Nothing}, varassoc::Vector{Int64}, mm_orig::ModelingToolkit.SparseMatrixCLIL{Int64, Int64})
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\alias_elimination.jl:265
 [6] alias_elimination(sys::ODESystem)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\alias_elimination.jl:12
 [7] structural_simplify(sys::ODESystem; simplify::Bool)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\abstractsystem.jl:857
 [8] structural_simplify(sys::ODESystem)
   @ ModelingToolkit C:\Users\xxxxxx\.julia\packages\ModelingToolkit\frHXN\src\systems\abstractsystem.jl:857
 [9] top-level scope
   @ REPL[53]:1
```

I also managed to get the same `AssertionError: r == 0` error after including a differential term:

```julia
using ModelingToolkit
const D = Differential(t)
@variables t
@variables v1(t) v2(t) i1(t) i2(t) i3(t)
eq = [
    v1 ~ sin(2pi*t)
    v1 - v2 ~ i1
    v2 ~ i2
    i1 ~ i2
    D(i3) ~ v1
]
@named sys = ODESystem(eq)
sys = structural_simplify(sys)
```

It's a DAE, but the source drives one differential term and another set of equations without differential terms.

--- a/src/systems/alias_elimination.jl
+++ b/src/systems/alias_elimination.jl
@@ -375,6 +375,8 @@ function locally_structure_simplify!(adj_row, pivot_col, ag, may_eliminate)
             d, r = divrem(alias_candidate[1], pivot_val)
             if r == 0 && (d == 1 || d == -1)
                 alias_candidate = -d => alias_candidate[2]
+            else
+                return false
             end
         end
         ag[pivot_col] = alias_candidate

--- a/src/systems/alias_elimination.jl
+++ b/src/systems/alias_elimination.jl
@@ -372,7 +372,10 @@ function locally_structure_simplify!(adj_row, pivot_col, ag, may_eliminate)
         # Note that when `nirreducible <= 1`, `alias_candidate` is uniquely
         # determined.
         if alias_candidate !== 0
-            alias_candidate = -exactdiv(alias_candidate[1], pivot_val) => alias_candidate[2]
+            d, r = divrem(alias_candidate[1], pivot_val)
+            if r == 0 && (d == 1 || d == -1)
+                alias_candidate = -d => alias_candidate[2]
+            end
         end
         ag[pivot_col] = alias_candidate
         zero!(adj_row)

--- a/test/nonlinearsystem.jl
+++ b/test/nonlinearsystem.jl
@@ -159,9 +159,21 @@ end
 
 # observed variable handling
 @variables t x(t) RHS(t)
-@parameters τ   
+@parameters τ
 @named fol = NonlinearSystem([0  ~ (1 - x)/τ], [x], [τ]; observed=[RHS ~ (1 - x)/τ])
 @test isequal(RHS, @nonamespace fol.RHS)
 RHS2 = RHS
 @unpack RHS = fol
 @test isequal(RHS, RHS2)
+
+# issue #1358
+@variables t
+@variables v1(t) v2(t) i1(t) i2(t)
+eq = [
+    v1 ~ sin(2pi*t)
+    v1 - v2 ~ i1
+    v2 ~ i2
+    i1 ~ i2
+]
+@named sys = ODESystem(eq)
+@test length(equations(structural_simplify(sys))) == 0

